source/agora/utils/Test.d

```diff
@@ -385,6 +385,41 @@ public struct TxBuilder
         return this;
     }
 
+    /***************************************************************************
+
+        Attaches to a range of tuples.
+
+        The tuple should have an `Output` and a `Hash` as its first and
+        second element, respectively.
+
+        Params:
+            rng = the range of tuple
+
+        Returns:
+            A reference to `this` to allow for chaining
+
+    ***************************************************************************/
+
+    public ref typeof(this) attach (RNG) (scope RNG rng)
+        @safe pure nothrow return
```

Comment:
Oh! A reference to `this` Nice!

Reply:
I noticed that all overloads of `attach` are using `ref typeof(this)`.